fs/memoryfs.py

```diff
@@ -254,7 +292,8 @@ def set_entry(self, name, dir_entry):
 
     def remove_entry(self, name):
         # type: (Text) -> None
-        del self._dir[name]
+        dirent = self._dir.pop(name)
+        dirent.clean_up()
```

Comment:
I don't think this is required. If there are no more references to the DirEntry the memory will be garbage collected anyway.